### DISCUSSION TITLE
Rename GL JS to Web?

### DISCUSSION THREAD

**Discussion by birkskyum** [maplibre/maplibre-gl-js#2643] [Polls]
Since the inception of the org, it has popped up from time to time if GL JS should be renamed Web or something else, because  GL JS was just the inherited name. This ticket explores if there is appeal to this in the community, and what the considerations around it is. Community feedback is very welcome.

**A case made for MapLibre GL / MapLibre GL JS**
* This name is similar to Mapbox GL JS, indicating ease of migration, similar feature set. It has served us well so far.
* We wouldn't have to do the work of renaming/rebranding.
* `maplibre-gl` npm package would keep it's name in either case, and it's most similar to MapLibre GL (JS)
* We can't control that all ecosystem packages are named maplibre/mapbox/map-gl-something, so that would also make a divergence.
* Our new css classes use the maplibregl naming.
* When pulling in the library from unpkg, the maplibregl object is created globally.

**A case made for MapLibre Web**
* As time goes on and Mapbox GL JS / MapLibre GL JS APIs do start to diverge, keeping the GL JS parallel going is becoming more of a source of confusion, as an identical public api and feature set is expected. By moving onto something else, and more intuitive, we stand on our own legs and take charge of our brand identity.

* It's cleaner, and 1 instead of 4 syllables.

* Alignment with Native and Style Spec
  * We recently remove the GL from the style spec.
  * We recently renamed GL Native to Native, to capture that it wasn't only OpenGL anymore. GL JS per v3 transitioned from WebGL to WebGL2, and will eventually end up with WebGPU, so it's quite similar.
  * We also transitioned from JS to TS per v2, so the situation around JS has changed to. It's also a bit confusing that the server SDK for NodeJS actually rely on Native, and not the "JS" version.
* We tend to use Web when casually referring to this library in discussions, but also in recruitment situations or even presentations can it be found.

**Comment by HarelM** [comment]
My two cents (personal opinion): there will be a lot of pain for people to migrate to the new package name, besides examples in SO, etc, and I don't see a real value in renaming (despite what's mentioned here).
Renaming just the repo will be even more confusing (much like other mapbox packages which doesn't have the exact repo name is always confusing to me).
While I like refactoring in general, I don't see real value here.

**Reply by birkskyum** [reply]
There is no change in package name or migration of users in question here - there was a separate discussion about scoped packaging [here](https://github.com/maplibre/maplibre-gl-js/discussions/2026), that concluded we don't want to put users through a migration.

**Reply by HarelM** [reply]
Can you clarify what is being suggested here? I'm confused...

**Reply by birkskyum** [reply]
I'm asking the community if the name of this product, in terms of branding, repo, readme etc should be updated, but with the npm package 'maplibre-gl' kept intact. It has been mentioned quite some times now, so I wanted to do a vibe check.
